src/python/bot/fuzzers/mutator_plugin.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Getting and using custom mutator plugins for libFuzzer."""
+
+import os
+
+from base import utils
+from google_cloud_utils import storage
+from system import archive
+from system import environment
+from system import shell
+
+MUTATOR_SHARED_OBJECT_FILENAME = 'mutator-plugin.so'
+ARCHIVES_SUBDIR_NAME = 'archives'
+PLUGINS_SUBDIR_NAME = 'plugins'
+
+
+def clear_plugins_directory():
+  shell.remove(environment.get_value('MUTATOR_PLUGINS_DIR'))
+
+
+def _get_mutator_plugins_bucket_url():
+  """Returns the url of the mutator plugin's cloud storage bucket."""
+  return 'gs://%s' % environment.get_value('MUTATOR_PLUGINS_BUCKET')
+
+
+def _get_mutator_plugins_subdir(subdir):
+  """Returns the path of the subdirectory named |subdir| in
+  MUTATOR_PLUGINS_DIR."""
+  return os.path.join(environment.get_value('MUTATOR_PLUGINS_DIR'), subdir)
+
+
+def _get_mutator_plugins_archives_dir():
+  """Returns path of the archives subdirectory in MUTATOR_PLUGINS_DIR"""
+  return _get_mutator_plugins_subdir(ARCHIVES_SUBDIR_NAME)
+
+
+def _get_mutator_plugins_unpacked_dir():
+  """Returns path of the unpacked plugins subdirectory in MUTATOR_PLUGINS_DIR"""
+  return _get_mutator_plugins_subdir(PLUGINS_SUBDIR_NAME)
+
+
+def _get_mutator_plugins_from_bucket():
+  """Returns list of the mutator plugin archives in the mutator plugin storage
+  bucket."""
+  return storage.list_blobs(_get_mutator_plugins_bucket_url())
+
+
+def _download_mutator_plugin_archive(mutator_plugin_archive):
+  """Downloads the |mutator_plugin_archive| from the mutator plugin storage
+  bucket to the plugin archives directory. Returns the path archive was
+  downloaded to."""
+  file_path = os.path.join(_get_mutator_plugins_archives_dir(),
+                           mutator_plugin_archive)
+  url = '%s/%s' % (_get_mutator_plugins_bucket_url(), mutator_plugin_archive)
+  assert storage.copy_file_from(url, file_path)
```

Comment:
We don't usually assert for errors like these, but instead log_error and continue.